File: ffmpegso/src/main/java/com/dastanapps/ffmpegso/VideoKit.java
package com.dastanapps.ffmpegso;

import processing.ffmpeg.videokit.CommandBuilder;
import processing.ffmpeg.videokit.LogLevel;
import processing.ffmpeg.videokit.VideoCommandBuilder;

/**
 * Created by Ilja Kosynkin on 06.07.2016.
 * Copyright by inFullMobile
 */
public class VideoKit {
    static {
        try {
            System.loadLibrary("ffmpegso");
        } catch (UnsatisfiedLinkError e) {
            e.printStackTrace();
        }
    }

    private LogLevel logLevel = LogLevel.NO_LOG;

    public void setLogLevel(LogLevel level) {
        logLevel = level;
    }

    public int process(String[] args) {
        return run(args);
    }

    // JNI
    public native String avformatinfo();

    public native String avcodecinfo();

    public native String avfilterinfo();

    public native String configurationinfo();

    public native int run(String[] args);

    public CommandBuilder createCommand() {
        return new VideoCommandBuilder(this);
    }
}

File: ffmpegso/src/main/java/com/dastanapps/ffmpegso/VideoKit.kt
package com.dastanapps.ffmpegso

import android.util.Log
import processing.ffmpeg.videokit.CommandBuilder
import processing.ffmpeg.videokit.VideoCommandBuilder

/**
 * Created by dastaniqbal on 19/07/2018.

 * 19/07/2018 11:02
 */
class VideoKit {
    init {
        try {
            System.loadLibrary("ffmpegso")
        } catch (e: UnsatisfiedLinkError) {
            e.printStackTrace()
        }
    }

    var videoKitListener: IVideoKit? = null

    external fun avformatinfo(): String

    external fun avcodecinfo(): String

    external fun avfilterinfo(): String

    external fun configurationinfo(): String

    external fun run(args: Array<String>): Int

    fun showProgress(progress: String) {
        Log.d("JNI:VidKit", progress)
        videoKitListener?.run {
            progress(progress)
        }
    }

    fun process(args: Array<String>): Int {
        return run(args)
    }

    fun createCommand(): CommandBuilder {
        return VideoCommandBuilder(this)
    }

    interface IVideoKit {
        fun progress(progress: String)
    }
}